diffgraph/ai_analysis.py

```
@@ -15,6 +19,28 @@ class DiffAnalysis(BaseModel):
 
+def exponential_backoff_retry(func):
+    """Decorator to implement exponential backoff retry logic."""
+    def wrapper(*args, **kwargs):
+        max_retries = 5
+        base_delay = 1  # Start with 1 second
+        max_delay = 60  # Maximum delay of 60 seconds
+
+        for attempt in range(max_retries):
+            try:
+                return func(*args, **kwargs)
+            except openai.RateLimitError as e:
```

Ruff (0.11.9)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)